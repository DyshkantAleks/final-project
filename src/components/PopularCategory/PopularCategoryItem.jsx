import React from "react";
import {device} from "../../styles/breakpoints/breakpoints";


import styled from 'styled-components';


export const PopularCategoryItem = (props) => {
    const {textForTitle, textForSubtitle, backGround} = props;

    return (
        <Item>
            <ImageContainer>
                <Image src={backGround}
                />
            </ImageContainer>
            <Title>{textForTitle}</Title>
            <Subtitle>{textForSubtitle}</Subtitle>
        </Item>

    )
};

const ImageContainer = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: -1;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
`;
const Image = styled.img`
width: 100%;
height: auto;
`

const Item = styled.section`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: flex-start;
    min-height: 15rem;
    padding: 1.5rem 2rem;
    color: white;
    //background-color: #A291A5;
    
   background-repeat: no-repeat;
   background-position: center;
   background-size: cover;
   
   position: relative;
    
    
    @media ${device.tabletM} {
     height: 20rem;
    }

@media ${device.tabletL} {
 height: 25rem;
}   
`;

const Title = styled.h3`
    font-size: 2.4rem;
      line-height: 1.5;
      margin: 0;
      font-weight: bold;
    
    
@media ${device.tabletM} {
  font-size: 3rem;
}
 
`;

const Subtitle = styled.h4`
    font-size: 1.2rem;
      max-width: 80%;
      margin: 1rem 0;
      text-align: left;
    
    
@media ${device.tabletM} {
      font-size: 1.6rem;
      max-width: 70%;
}
@media ${device.desktop} {
      max-width: 60%;
}  
 
`;