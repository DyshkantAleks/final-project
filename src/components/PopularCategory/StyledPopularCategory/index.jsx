import styled from "styled-components";
import {device} from "../../../styles/breakpoints/breakpoints";

export const List = styled.div`
     display: grid;
    grid-template-columns: 1fr;
    gap: 2rem 2rem;
   
@media ${device.tabletM} {
 grid-template-columns: 1fr 1fr;
}

`;

export const Item = styled.section`
display: flex;
flex-direction: column;
justify-content: flex-end;
align-items: flex-start;
min-height: 15rem;
padding: 1.5rem 2rem;
background-repeat: no-repeat;
background-position: center;
background-size: cover;
position: relative;
transition: all 0.5s ease;
&:hover {
transform: scale(1.01);
box-shadow: -1px -1px 18px 0 #6b7280;
}

@media ${device.tabletM} {
height: 20rem;
 }

@media ${device.tabletL} {
 height: 25rem;

}
`;


export const StaticImageContainer = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: -1;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
`;
export const StaticImage = styled.img`
width: 100%;
height: auto;
`;


export const StaticTitle = styled.h3`
    font-size: 2.4rem;
      line-height: 1.5;
      margin: 0;
      font-weight: bold;
      text-shadow: 0 0 black;
      color: white;
    
    
@media ${device.tabletM} {
  font-size: 3rem;
}
 
`;

export const StaticSubtitle = styled.h4`
    font-size: 1.2rem;
      max-width: 80%;
      margin: 1rem 0;
      text-align: left;
      text-shadow: 0 0 black;
      color: white;
    
    
@media ${device.tabletM} {
      font-size: 1.6rem;
      max-width: 70%;
}
@media ${device.desktop} {
      max-width: 60%;
}  
 
`;


export const HoverImageContainer = styled.div`
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    z-index: -1;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
    opacity: 0.3;
    
`;

export const HoverTitle = styled.h3`
    font-size: 2.4rem;
      line-height: 1.5;
      margin: 0;
      font-weight: bold;
      //text-shadow: 0 0 black;
      color: black;
    
    
@media ${device.tabletM} {
  font-size: 3rem;
}
 
`;

export const HoverSubtitle = styled.h4`
    font-size: 1.2rem;
      max-width: 80%;
      margin: 1rem 0;
      text-align: left;
      //text-shadow: 0 0 black;
      color: black;
    
    
@media ${device.tabletM} {
      font-size: 1.6rem;
      max-width: 70%;
}
@media ${device.desktop} {
      max-width: 60%;
}  
 
`;