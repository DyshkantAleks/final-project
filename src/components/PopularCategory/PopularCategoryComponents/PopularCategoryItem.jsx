import React, { useState } from 'react';
import PropTypes from 'prop-types';

import { Item } from '../StyledPopularCategory';
import { StaticCategoryItem } from './HoverItem/StaticCategoryItem';
import { HoverCategoryItem } from './HoverItem/HoverCategoryItem';

export const PopularCategoryItem = (props) => {
  const { textForTitle, textForSubtitle, backGround } = props;
  const [hover, setHover] = useState(false);
  const onMouseEnterHandler = () => { setHover(true) };
  const onMouseLeaveHandler = () => { setHover(false) };

  return (
    <Item onMouseEnter={onMouseEnterHandler} onMouseLeave={onMouseLeaveHandler}>
      {hover && (
        <HoverCategoryItem
          textForTitle={textForTitle} textForSubtitle={textForSubtitle}
          backGround={backGround}
        />
      )}
      {!hover && (
        <StaticCategoryItem
          textForTitle={textForTitle} textForSubtitle={textForSubtitle}
          backGround={backGround}
        />
      )}
    </Item>
  )
};

PopularCategoryItem.defaultProps = {
  textForTitle: 'Заголовок',
  textForSubtitle: 'Текст',
  backGround: 'https://res.cloudinary.com/dg-home/image/upload/v1605196877/General/Categories/popular-category-table-min_fmalgj.png'
};

PopularCategoryItem.propTypes = {
  textForTitle: PropTypes.string,
  textForSubtitle: PropTypes.string,
  backGround: PropTypes.string
};