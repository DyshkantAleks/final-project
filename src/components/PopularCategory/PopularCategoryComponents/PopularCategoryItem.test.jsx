import React from 'react';
import { shallow } from 'enzyme';

import { PopularCategoryItem } from './PopularCategoryItem';
import { Item } from '../StyledPopularCategory';

const setUp = (props) => shallow(<PopularCategoryItem {...props} />)

describe('PopularCategoryItem component', () => {
  let component;
  beforeEach(() => {
    component = setUp()
  });
  it('should be render without crushing', () => {
      const wrapper = component.find(Item);
      expect(wrapper.length).toEqual(1);
  });
  it('snaphot of <PopularCategoryItem />', () => {
    const wrapper = component.find(Item);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
