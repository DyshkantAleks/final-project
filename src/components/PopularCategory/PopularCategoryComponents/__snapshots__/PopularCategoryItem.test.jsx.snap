// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularCategoryItem component snaphot of <PopularCategoryItem /> 1`] = `
"<styled.section onMouseEnter={[Function: onMouseEnterHandler]} onMouseLeave={[Function: onMouseLeaveHandler]}>
  <StaticCategoryItem textForTitle=\\"Заголовок\\" textForSubtitle=\\"Текст\\" backGround=\\"https://res.cloudinary.com/dg-home/image/upload/v1605196877/General/Categories/popular-category-table-min_fmalgj.png\\" />
</styled.section>"
`;
